lib/syskit/process_managers/remote/server/process.rb

```diff
@@ -250,14 +266,28 @@ def spawn
                         @ior_message = +""
 
                         @ior_read_fd, ior_write_fd = IO.pipe
+                        control_read_fd, @control_write_fd = IO.pipe
```

Comment:
Why does the `rr` execution mode implementation need to use the control fd instead of signal handling?

Reply:
It is discussed in the control fd implementation

Essentially, support for signals in rr is sketchy. But, to be honest, signals are really not that great in multithreaded code.

Reply:
> signals are really not that great in multithreaded code.

I figured, just wondered if there was any special requirements for rr.